gr-utils/python/blocktool/core/iosignature.py

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright 2019 Free Software Foundation, Inc.
+#
+# This file is part of GNU Radio
+#
+# GNU Radio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3, or (at your option)
+# any later version.
+#
+# GNU Radio is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with GNU Radio; see the file COPYING.  If not, write to
+# the Free Software Foundation, Inc., 51 Franklin Street,
+# Boston, MA 02110-1301, USA.
+#
+""" Module to get io_signature of the header block """
+
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import itertools
+import logging
+import string
+
+from blocktool.core import Constants
+
+LOGGER = logging.getLogger(__name__)
+
+
+def io_signature(io_file):
+    """
+    function to generate the io_signature of the block
+    : returns the io parmaters
+    """
+    parsed_io = {
+        "input": {
+            "signature": None
+        },
+        "output": {
+            "signature": None
+        }
+    }
+    with open(io_file, 'r') as impl:
+        lines = []
+        for line in impl:
+            if Constants.IO_SIGNATURE in line:
+                lines.append(line)
+    impl.close()
```

Comment:
When using `with open`, the file gets automatically closed after the `with` block thanks to the context manager, so line 54 is not required.
```suggestion
```